apps/artemis-web/middleware/limiter.ts

```diff
@@ -0,0 +1,57 @@
+import rateLimit from "express-rate-limit";
+require('dotenv').config();
+
+
+function calculateNextResetTime(windowMs) {
+    const d = new Date();
+    d.setMilliseconds(d.getMilliseconds() + windowMs);
+    return d;
+}
+
+function MemoryStore(windowMs, file) {
+    let hits = {};
+    let resetTime = calculateNextResetTime(windowMs);
+
+    this.incr = function (key, cb) {
+        if (hits[file + "," + key]) {
+            hits[file + "," + key]++;
+        } else {
+            hits[file + "," + key] = 1;
+        }
+
+        cb(null, hits[file + "," + key], resetTime);
+    };
+
+    this.decrement = function (key) {
+        if (hits[file + "," + key]) {
+            hits[file + "," + key]--;
+        }
+    };
+
+    // export an API to allow hits all IPs to be reset
```

Comment:
I see that these are also exported as API calls. Are they protected? :) Can they be done only by authenticated/authorized users or everyone?